build-aux/io.github.flattool.Ignition.json

```diff
@@ -0,0 +1,65 @@
+{
+    "id" : "io.github.flattool.Ignition",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "47",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.node20"
+    ],
+    "command" : "io.github.flattool.Ignition",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=xdg-config/autostart:create",
+        "--filesystem=xdg-data/icons:ro",
+        "--filesystem=xdg-data/flatpak:ro",
```

Comment:
these 3 need exceptions. why are they required?

Reply:
you can run the manifest check https://docs.flathub.org/docs/for-app-authors/submission#run-the-linter to get the error codes, then open a pr with the exception https://docs.flathub.org/docs/for-app-authors/linter#exceptions

Reply:
The `xdg-config/autostart:create` item is so that my app can create and edit the user's autostart entries, as this location is compliant with the Freedesktop spec.
The `xdg-data/icons:ro` item is so that my app can load any icon themes applied by the user. This is more important for the following...
The `xdg-data/flatpak:ro` item is so that my app can see what graphical Flatpaks are installed on the system by seeing the list of .desktop entries.

Essentially my app needs to both be able to see all the .desktop entries located in the standard locations, to present the user with their list of installed apps they can select to autostart, and present the icons associated with these apps, as just presenting text would not be a great experience.

I will open a PR on the linter with a more eloquent explanation there. Thank you.

Reply:
Alright, the PR to the linter has been opened [here](https://github.com/flathub-infra/flatpak-builder-lint/pull/549)